src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,18 +146,16 @@ public void Start(PushRuntimeSettings limitations)
 
                     await concurrencyLimiter.WaitAsync(cancellationToken).ConfigureAwait(false);
 
+                    var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationTokenSource.Token);
```

Comment:
I don't think this is right, from what I understand here, `cancellationTokenSource.Token` is cancelled when the endpoint is stopping, which means we finish executing all in-flight messages, but on the other hand if a cancellation is triggered as part of the pipeline execution, we then probably want to abort the transaction (or not acknowledge the message)